cmds/core/timeout/main.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2012-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Run a command and kill it if it runs more than a specified duration
+//
+// Synopsis:
+//	timeout [-v] [-t duration-string] command [args...]
+//
+// Description:
```

Comment:
Does it make sense to document somewhere that a non-zero return code is not passed through exactly from the command being executed?   e.g. if the command returns '10' as the exit code, this program will return exit code '1' ?

Also, maybe document something about the environment variables handling?

Reply:
are there any environment variables handled specially?

Reply:
This now exits with the exit value of the command. 

Can't do much it was 1 :-)

Reply:
And the coverage drops ... because the command, in the end, is just very small.
It's a bit frustrating that the Go standard package doesn't have os.ExitErr or that we don't a log.FatalErr that
would unpack the error to its exit code. I've no idea how to handle this ... except, possibly, by passing the errorhandler into run as a function, which could be just os.Exit or something used in testing.

Reply:
ok, see what you think. I am setting an exitcode in the cmd struct now.

Reply:
we may want to do similar things in the backoff command. 

The backoff command, I just remembered, is the command in which coverage was improved and the command stopped working.

Reply:
> ok, see what you think. I am setting an exitcode in the cmd struct now.

I think setting cmd.exitcode is setting a trap for new code in the future that will forget to set it when a new error is returned and the default cmd.exitcode value '0' will be used.   

I can think of 2 naive solutions (naive because I haven't tested them):
1) Change "run" to return an int status code, this will force the author of any future "return" statements to think about what code to return
2) (Maybe preferred?) create a function that can be called (and tested) to return the status code from an error
```
func exitCode(err error) int {
  if err == nil {
    return 0
  }
  var e *exec.ExitError
  if errors.As(err, &e) {
    return e.ExitCode()
  }
  return 1
}
```

Reply:
but consider that this is no different than what exec does today. Lots of things do so. I thought of returning (errno, error) -- basically what syscall does, so I'm happy to do that too. I thought of that helper function, as well, but you can't easily set up exec.ExitError, it holds an os.ProcessState, which is basically all private -- no test possible. in fact note that exec.Cmd holds the process state for the exiting process, so that was my precedent. It's pretty common, actually: in fact os.ProcessState can old hold the stderr from the process. I suspect the code in this form is ok. If people forget to set things, well, that means they did not write the test.

Reply:
done